js/abp-preprocessed.es6.js

```diff
@@ -68,3 +71,28 @@ chrome.alarms.onAlarm.addListener(alarm => {
 // set an alarm to recheck the lists
 // update every 3 hours
 chrome.alarms.create('updateEasyLists', {periodInMinutes: 180})
+
+// add version param to url on the first install and
+// only once a day after than
+function getVersionParam () {
+    let version = settings.getSetting('version') || 'v1'
+    let lastUpdate = settings.getSetting('lastUpdate')
+    let now = Date.now()
+    let versionParam
+
+    // check delta for last update or if lastUpdate does
+    // not exist then this is the initial install
+    if (lastUpdate) {
+        let delta = now - new Date(lastUpdate)
+            
+        if (delta > ONEDAY) {
+            versionParam = `&v=${version}`
+        }
+    } else {
+        versionParam = `&v=${version}`
+    }
+            
+    settings.updateSetting('lastUpdate', now)
+
+    return versionParam
+}
```

Comment:
I like the idea of this. The naming seems a little off tho? We aren't actually tracking version here (`versionParam` would be a name for something that would track against the version of the extension in the FF store, for example). To me, this should be named `lastUpdateParam`.

That said, I also like the idea of a `versionParam` that actually tracks the version number (from the store). Would it be possible to add that here? Might simplify bug reporting and help us identify update errors via kibana, etc.

Reply:
Wait, sorry, I think I misread this... hold on.

Reply:
Okay, just read the code again and I think there is a subtle bug (either that or I am misunderstanding something(!) -- which could also be possible :) )

Within this file, `getVersionParam()` sets the `lastUpdate` setting and is called by `updateLists()`, which fires once automatically on browser start, and is subsequently called by a chrome alarm that is set to run every 180 minutes. Every time `getVersionParam()` is run -- **_no matter what_** -- you are setting `settings.updateSetting('lastUpdate', now)` where `now` is `Date.now()` (line 95).

The logic just above this on line 88 says: If our setting `lastUpdate` is more than 24hrs old, then append the `v=${version}` param. But if we are calling `updateLists()`, which calls `getVersionParam()` every 180 minutes and is always updating `lastUpdate` setting every time no matter what, then we'll never get to the `delta > ONEDAY` state. 

**This will result in never sending the version param except on browser startup, and a dramatic undercollection of Daily Active User (DAU) pings.**

Reply:
As a sidenote, I still think setting name should be more specific than `lastUpdate`. Perhaps a more specific name that cannot be confused for other things would be: `lastEasyListUpdate`.